sotodlib/site_pipeline/update_det_match.py

```diff
@@ -0,0 +1,374 @@
+    for detset in np.unique(aman.det_info.detset):
+        if detset not in finished_detsets:
+            new_detsets.append(detset)
+
+    logger.info(f"Loaded obs_id {obs_id}. Running matches for detsets:")
+    for ds in new_detsets:
+        logger.info(f"    - {ds}")
+
+    for ds in new_detsets:
+
+        stream_id = aman.det_info.stream_id[aman.det_info.detset == ds][0]
+        # Try to get wafer slot info from book idx
+        if 'wafer_slots' in book_idx:
+            for ws in book_idx['wafer_slots']:  
+                if ws['stream_id'] == stream_id:
+                    wafer_slot = ws['wafer_slot']
+                    break
+            else:
+                logger.error(
+                    f"Could not find wafer_slot from book index for ds={detset}, "
+                    f"obs_id={obs_id}"
+                )
+                raise Exception("Could not find wafer-slot")
+        else:
+            wafer_slot = None
+
+        match = run_match_aman(runner, aman, ds, wafer_slot=wafer_slot)
+        fpath = os.path.join(runner.match_dir, f"{ds}.h5")
+        match.save(fpath)
+        logger.info(f"Saved match to file: {fpath}")
+
+    return aman
+
+
+def scan_for_freq_offset(rs0, rs1, freq_offsets, match_pars=None, show_pb=True):
+    """
+    Scans through a list of frequency offsets to find optimal match between two
+    res-sets.
+
+    Returns
+    ----------
+    costs : np.ndarray
+        Costs of size ``len(freq_offsets)`` containing the matching cost
+        at each frequency
+    opt_freq : float
+        Optimal offset-frequency for match.
+    """
+    if match_pars is None:
+        match_pars = det_match.MatchParams()
+    else:
+        match_pars = deepcopy(match_pars)
+    freq_offsets = np.array(freq_offsets)
+    
+    rs0 = deepcopy(rs0) 
+    rs1 = deepcopy(rs1)
+
+    costs = np.full_like(freq_offsets, np.nan)
+    for i, offset in enumerate(tqdm(freq_offsets, disable=(not show_pb))):
+        match_pars.freq_offset_mhz = offset
+        match = det_match.Match(rs0, rs1, match_pars=match_pars)
+        costs[i] = match.matching_cost
+
+    imin = np.argmin(costs)
+    opt_freq = freq_offsets[imin]
+    logger.info(f"Found freq offset: {opt_freq}")
+    return costs, opt_freq
+
+
+def update_manifests(runner: Runner, detset):
+    det_match_idx = os.path.join(runner.cfg.results_path, 'det_match.sqlite')
+    det_match_h5 = os.path.join(runner.cfg.results_path, 'det_match.h5')
+    assignment_idx = os.path.join(runner.cfg.results_path, 'assignment.sqlite')
+    assignment_h5 = os.path.join(runner.cfg.results_path, 'assignment.h5')
+
+    match_file = os.path.join(runner.cfg.results_path, f'matches/{detset}.h5')
+    with h5py.File(match_file, 'r') as f:
+        ra = np.array(f['merged'])
+
+    names = list(ra.dtype.names)
+    idx = names.index('readout_id')
+    names[idx] = 'dets:readout_id'
+    ra.dtype.names = tuple(names)
+    assignment = ra[['dets:readout_id', 'det_id']]
```

Comment:
In the `assignment` output, which is intended for det_info, the field must be called `dets:det_id`.  You'll find that with those two columns, it won't load as standard metadata anymore.  But it will load if you set `det_info: true` (and it will populate `det_info.det_id`).